src/exif/tests/service.rs

```diff
@@ -0,0 +1,159 @@
+use crate::exif::process_single_file_exif;
+use crate::model::file_types::FileTypes;
+use crate::model::repository::FileRecord;
+use crate::repository::{file_repository, open_connection};
+use crate::service::file_service::file_dir;
+use crate::test::{cleanup, init_db_folder};
+use rocket::tokio;
+
+#[cfg(not(ci))]
+#[tokio::test]
+async fn successfully_parsing_exif_data_stores_in_db() {
+    init_db_folder();
+    // Copy a test file with EXIF data from test_assets
+    std::fs::copy("test_assets/test.png", format!("{}/test.png", file_dir()))
+        .expect("Failed to copy test file to the file directory");
+    
+    // Create a file record with a hard-coded old date
+    let old_date = chrono::NaiveDateTime::parse_from_str("0001-01-01 00:00:00", "%Y-%m-%d %H:%M:%S").unwrap();
+    let file_record = FileRecord {
+        id: None,
+        name: "test.png".to_string(),
+        parent_id: None,
+        create_date: old_date,
+        size: 100,
+        file_type: FileTypes::Image,
+    }.save_to_db();
+
+    let file_id = file_record.id.unwrap();
+
+    // Verify the initial date
+    let con = open_connection();
+    let initial_record = file_repository::get_file(file_id, &con).unwrap();
+    con.close().unwrap();
+    assert_eq!(initial_record.create_date, old_date, "Initial date should be the old date");
+
+    // Process the file
+    let res = process_single_file_exif(file_id.to_string()).await;
+    assert!(
+        res,
+        "process_single_file_exif should return true on success"
+    );
+
+    // Verify the date was updated to something greater than the old date
+    let con = open_connection();
+    let updated_record = file_repository::get_file(file_id, &con).unwrap();
+    con.close().unwrap();
+
+    assert!(
+        updated_record.create_date > old_date,
+        "Date should be updated to be greater than the old date"
+    );
+
+    cleanup();
+}
+
+#[tokio::test]
+async fn failing_to_parse_exif_stores_current_date() {
+    init_db_folder();
+    // Create a file without EXIF data - use an image extension but with non-image content
+    let file_content = "This is not an image file";
+    std::fs::write(format!("{}/test.jpg", file_dir()), file_content)
```

Comment:
Use test functions to create files in the file system and db instead of manually calling std fs